IntegrationTests/UITests/CommandsDialogs/FormInitTests.cs

```diff
@@ -112,10 +112,10 @@ private void RunFormTest(Action<FormInit> testDriver, string path)
 
         private void RunFormTest(Func<FormInit, Task> testDriverAsync, string path)
         {
-            UITest.RunForm(
-                () =>
+            UITest.RunDialog(
```

Comment:
❓ Why can this not use `RunForm`?

Reply:
Because `Form.Show` should be called by a UI thread. We don't have a message loop yet when executing tests.

Reply:
Have you seen a particular case where it results in an error? If so, what error message are you seeing? The UI thread state for tests is initialized here, so we shouldn't ever have problems with a message loop:

https://github.com/gitextensions/gitextensions/blob/b46d08c19282a81a70cf2caa15f9611afd72dea0/UnitTests/CommonTestUtils/ConfigureJoinableTaskFactoryAttribute.cs#L45-L50

Reply:
`Application.MessageLoop` is `false` always.
I haven't seen an according error message. Though these tests fail often and there is no single cause.

Reply:
> Application.MessageLoop is false always.

This seems fine. We don't use this property in a way that would impact these tests. I'd stick with the existing `RunForm` unless we have a specific reason to believe this is the cause of a test failure.

Reply:
We have several types of test hangs. A quite often one is FormCommit visible without progress.
This was most likely resolved by
- `UITest.RunDialog` or
- `Application.DoEvents` before and after `WaitForPendingOperations` or
- the combination of both.

[FormCommit hang:](https://ci.appveyor.com/project/gitextensions/gitextensions/builds/30271214)
```
Error : GitExtensions.UITests.CommandsDialogs.FormCommitTests.Should_display_detached_head_info_in_statusbar
System.OperationCanceledException : The operation was canceled.
   at System.Threading.CancellationToken.ThrowOperationCanceledException()
   at System.Threading.CancellationToken.ThrowIfCancellationRequested()
   at Microsoft.VisualStudio.Threading.ThreadingTools.<WithCancellationSlow>d__12.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.VisualStudio.Threading.JoinableTaskCollection.<JoinTillEmptyAsync>d__20.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at GitUI.ThreadHelper.<JoinPendingOperationsAsync>d__14.MoveNext() in C:\projects\gitextensions\GitExtUtils\GitUI\ThreadHelper.cs:line 118
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.Run(Func`1 asyncMethod, JoinableTaskCreationOptions creationOptions, Delegate entrypointOverride)
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.Run(Func`1 asyncMethod)
   at CommonTestUtils.AsyncTestHelper.WaitForPendingOperations(CancellationToken cancellationToken) in C:\projects\gitextensions\UnitTests\CommonTestUtils\AsyncTestHelper.cs:line 45
   at CommonTestUtils.AsyncTestHelper.WaitForPendingOperations(TimeSpan timeout) in C:\projects\gitextensions\UnitTests\CommonTestUtils\AsyncTestHelper.cs:line 39
   at GitExtensions.UITests.CommandsDialogs.FormCommitTests.<>c.<Should_display_detached_head_info_in_statusbar>b__8_0(FormCommit form) in C:\projects\gitextensions\IntegrationTests\UITests\CommandsDialogs\FormCommitTests.cs:line 113
   at GitExtensions.UITests.CommandsDialogs.FormCommitTests.<>c__DisplayClass20_0.<RunFormTest>b__0(FormCommit form) in C:\projects\gitextensions\IntegrationTests\UITests\CommandsDialogs\FormCommitTests.cs:line 317
   at GitExtensions.UITests.UITest.<>c__DisplayClass1_0`1.<<RunForm>b__0>d.MoveNext() in C:\projects\gitextensions\IntegrationTests\UITests\UITest.cs:line 44
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at Microsoft.VisualStudio.Threading.JoinableTask.<JoinAsync>d__68.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   at Microsoft.VisualStudio.Threading.JoinableTaskFactory.Run(Func`1 asyncMethod, JoinableTaskCreationOptions creationOptions, Delegate entrypointOverride)
   at Microsoft.VisualStudio.Threading.JoinableTask.Join(CancellationToken cancellationToken)
   at GitExtensions.UITests.UITest.RunForm[T](Action showDialog, Func`2 runAsync) in C:\projects\gitextensions\IntegrationTests\UITests\UITest.cs:line 56
   at GitExtensions.UITests.CommandsDialogs.FormCommitTests.RunFormTest(Func`2 testDriverAsync, CommitKind commitKind) in C:\projects\gitextensions\IntegrationTests\UITests\CommandsDialogs\FormCommitTests.cs:line 346
   at GitExtensions.UITests.CommandsDialogs.FormCommitTests.RunFormTest(Action`1 testDriver, CommitKind commitKind) in C:\projects\gitextensions\IntegrationTests\UITests\CommandsDialogs\FormCommitTests.cs:line 320
   at GitExtensions.UITests.CommandsDialogs.FormCommitTests.Should_display_detached_head_info_in_statusbar() in C:\projects\gitextensions\IntegrationTests\UITests\CommandsDialogs\FormCommitTests.cs:line 126
```

Reply:
The hang is most likely caused by the test trying to execute synchronously. Instead of calling `WaitForPendingOperations` and blocking on the result, it needs to await the result of an asynchronous wait for pending operations.

Reply:
> Have you seen a particular case where it results in an error?

https://github.com/gitextensions/gitextensions/pull/7625#issuecomment-578446428

Reply:
Thanks, I'll look into that case

Reply:
@mstv  I ran the test in question 100 times locally and was not able to observe a failure. The test used `UITest.RunForm`, and passed 100 of 100 times.

Reply:
Try it with https://github.com/gitextensions/gitextensions/commit/59d118a9d6e7398530022ce6df34dff1a2848a1b or https://github.com/gitextensions/gitextensions/commit/bb41709442f93a7755b61e2b6a0a7f7cbc82d3c4 in my fork, please.

Reply:
With https://github.com/gitextensions/gitextensions/commit/bb41709442f93a7755b61e2b6a0a7f7cbc82d3c4 it fails.
Without these `Application.DoEvents`, many tests fail because of timeouts.

Reply:
The hang should be resolved by #7691 and #7692. I do not believe we need to increase complexity by introducing a `RunDialog` method.

Reply:
> I do not believe we need to increase complexity by introducing a RunDialog method.

At least not yet. It may still be necessary, but I'm not all the way convinced yet.

Reply:
During my work on #7691, I determined _with certainty_ that the `RunDialog` method is not required. NUnit will automatically [start the application message pump](https://github.com/nunit/nunit/blob/30df2d13c0f67eeb6b3dc7f56eebd7c5cfb5c7de/src/NUnitFramework/framework/Internal/MessagePumpStrategy.cs#L94) the first time an [asynchronous test yields](https://github.com/nunit/nunit/blob/30df2d13c0f67eeb6b3dc7f56eebd7c5cfb5c7de/src/NUnitFramework/framework/Internal/AsyncToSyncAdapter.cs#L54-L58).